File: LeetCode/Java/0283. Move Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int nonZero = 0;
        
        for (int i = 0; i < nums.length; i++) 
            if (nums[i] != 0)
                swap(nums, i, nonZero++);
    }
    
    private void swap(int[] nums, int i, int j) {
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

File: LeetCode/Python/0283. Move Zeroes.py
class Solution:
    def moveZeroes(self, arr: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(arr)
        non_zero = 0
        
        for i in range(n):
            if arr[i] != 0:
                arr[i], arr[non_zero] = arr[non_zero], arr[i]
                non_zero += 1
